src/unix.rs

```diff
@@ -123,6 +149,18 @@ impl Session {
             .expect("driver task has gone away");
         Perform { inner: rx }
     }
+
+    pub fn perform2<H: Handler + Send>(&self, handle: Easy2<H>) -> Perform2<H> {
+        let (tx, rx) = oneshot::channel();
+        self.tx
+            .borrow_mut()
+            .unbounded_send(Message::Execute2(unsafe { mem::transmute(handle) }, tx))
```

Comment:
I originally tried that, but unfortunately it didn't work well. Apart from the verbosity and inefficiency of boxing, it also required a manual downcast implementation on `Handler` because we can only get a reference out of `Easy2` which doesn't provide any downcast methods.